packages/block-editor/src/components/link-control/index.js

```diff
@@ -326,9 +326,9 @@ function LinkControl( {
 	 * the next render, if focus was within the wrapper when editing finished.
 	 */
 	function stopEditing() {
-		isEndingEditWithFocus.current =
-			!! wrapperNode.current &&
-			wrapperNode.current.contains( document.activeElement );
+		isEndingEditWithFocus.current = wrapperNode.current?.contains(
```

Comment:
I committed your suggestion. 

How about adding these findings to [the JavaScript section of our coding guideline](https://github.com/WordPress/gutenberg/blob/master/docs/contributors/coding-guidelines.md#javascript)?

I simply thought optional chaining is just a good way to remove `&&` and check validness of variables. But there are many pitfalls we have to avoid. These things are important but cannot be easily found in other documentations. 

I think it's good for us and other future developers.

Reply:
@sainthkh For sure, I think it makes sense to be be documenting these. I'd actually planned to bring some of these up in next week's JavaScript chat, moreso as observations of common issues people may encounter, but documentation is a good action item. I think we're still so early on in its usage that we're still at a point of identifying these issues. That said, the coding guidelines are always a "working document" and be revised over time. 

As with the other discussion, I think we'll ultimately want to come to rely on our tooling:

- #21984 is a good example where ESLint can alert us against potential errors.
- This is a good example where TypeScript type checking can alert us about assigning an invalid type (a non-boolean)†

† Aside: I actually tried to test this in code by adding `@ts-check` to the top of the file, and it didn't exactly work as I expected. Not sure if it's an issue with JavaScript typing, with typing of `useRef`, or other. Maybe something we need to figure out in the future, or if it's an upstream issue, to wait for a fix.

Reply:
I created a pull request at #22029 to document the two gotchas observed here.